contributing/contributing-code.md

```diff
@@ -0,0 +1,115 @@
+- [ ] Follow the parts of the [Implementation](#implementation) section that makes sense for your change.  There are many parts of the process that you probably won't need to follow for a minor fix--e.g. you may not need to make validator changes or put your change behind an experiment for minor fixes.  If in doubt ask your reviewer.
+
+### Process for significant changes
+
+Significant changes (e.g. new components or significant changes to behavior) require consultation with and approval from knowledgeable members of the community.
+
+- [ ] *Before you start coding*, [find a reviewer](#find-a-reviewer) who you can discuss your feature with and who can help guide you through the process.
+- [ ] File an [Intent-to-implement (I2I)](TODO) GitHub issue and cc your reviewer on it.  The I2I should include:
+  -  A high-level description of the feature.
+  -  A description of the API you plan to create.
+  -  If you are integrating a third-party service, provide a link to the third-party's site and product.
+  -  Details on any data collection or tracking that the feature might perform.
+  -  A prototype or mockup (for example, an image, a GIF, or a link to a demo).
+- [ ] Determine who needs to approve your I2I.  Changes that have a significant impact on AMP's behavior or significant new features require the approval from the Approvers Working Group (WG).  Work with your reviewer to determine whether the feature is a significant enough change that it requires approval from the Approvers Working Group and/or any other Working Group.
+- [ ] Get pre-approval from the Approvers WG if needed.  For features that require approval from the Approvers WG, at least 3 members of the Approvers Working Group should provide pre-approval on the I2I before significant implementation work proceeds.
+- [ ] Finalize the design of your feature.
+  - Familiarize yourself with our [Design Principles](DESIGN_PRINCIPLES.md).
+  - Your reviewer can help you determine if your feature requires a design doc and whether it should be brought to a [design review](./design-reviews.md).
+- [ ] Proceed with the [implementation](#implementation) of your feature.
+- [ ] For features that require approval from the Approvers WG, file an [Intent-to-ship (I2S) issue](TODO).  Indicate which experiment is gating the feature and a rollout plan.  Once this issue is approved by 3 members of the Approver WG the rollout plan described in the I2S may proceed.
+
+## Find a reviewer
+
+If you're making a minor fix and just want to find someone to review/approve your code, look at the OWNERS.yaml files in the directories where you're modifying code and assign your PR to one of them (or cc them on the PR).
+
+A reviewer is needed if you are making a more substantial change to AMP.  A reviewer is a member of the AMP community who is knowledgeable about the area you are modifying and who can guide you from the design phase all the way through launch.
+
+To find a reviewer:
```

Comment:
Only folks with write access can actually assign reviewers. Should we document what to do when that isn't available? @-mention people?